jujugui/static/gui/src/app/components/button-dropdown/button-dropdown.js

```diff
@@ -0,0 +1,142 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+
+'use strict';
+
+/**
+  Creates a component that has an icon, which when clicked, opens a dropdown
+  menu with list items.
+*/
+class ButtonDropdown extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      showDropdown: false
+    };
+  }
+
+  /**
+    Passed into the dropdown component to call when the user clicks outside
+    of it. We use this trigger to close the dropdown.
+    @param {Object} e The click event.
+  */
+  _handleDropdownClickOutside(e) {
+    // If they click the button again we don't want it to clse the menu in the
+    // clickoutside as the _toggleDropdown will handle that.
+    if (!ReactDOM.findDOMNode(this).contains(e.target)) {
+      this.setState({showDropdown: false});
```

Comment:
s/showDropdown/menuDisplayed/ so that it's more apparent that it's a bool? Also, that would avoid repeating dropdown. [shrug]